File: tier1/csharp/f5e71727/Report.cs
using System.IO;

public static class Report
{
    private sealed class Request
    {
        public string Name { get; set; }
    }

    public static string Build(string name)
    {
        var request = new Request();
        request.Name = name;
        return Read(request);
    }

    private static string Read(Request request)
    {
        return File.ReadAllText("/srv/reports/" + request.Name);
    }
}

File: tier1/java/236f8d7e/Report.java
import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Files;
import java.nio.file.Paths;

public final class Report {
    private static final class Request {
        private String name;
        void setName(String value) {
            this.name = Paths.get(value).getFileName().toString();
        }
        String getName() {
            return this.name;
        }
    }

    public static String build(String name) {
        Request request = new Request();
        request.setName(name);
        return read(request);
    }

    private static String read(Request request) {
        try {
            Path base = Paths.get("/srv/reports").toRealPath();
            Path target = base.resolve(request.getName()).normalize();
            if (!target.startsWith(base)) {
                return "";
            }
            return new String(Files.readAllBytes(target));
        } catch (IOException e) {
            return "";
        }
    }
}
